### PR TITLE
Improve jsdoc

### LINKED ISSUE #178: Make the generated functions/options more better
I want to improve the quality of automatically generated functions/options.

## Fix function signature

- [x] #179
      Some functions are out of range.
      e.g.) [`assert_true()`](https://deno.land/x/denops_std@v4.1.1/function/vim/mod.ts?s=assert_true)
- [x] #181
      e.g.) [`buffer_exists()`](https://deno.land/x/denops_std@v4.1.1/function/vim/mod.ts?s=buffer_exists) or [`strpbrk()`](https://deno.land/x/denops_std@v4.1.1/function/mod.ts?s=strpbrk).     

## Fix option signature

- [x] #183
      A lot of options are missing.

## Improve JSDoc

- [ ] **Replace tab to spaces**:
      Indentation is broken when tab characters are included.
- [ ] **Keep blank lines**:
      All blank lines in functions are gone.
- [ ] **Keep trailing `>`**:
      The trailing `>` like `<CR>` has been removed.
- [ ] **Keep tags in sentences**:
      All tags are gone.
- [ ] **Convert code block**:
      Code blocks in help are indented unevenly.
- [ ] **Remove header mark `~`**:
- [ ] **Replace nvim list marker**:
- [ ] **Convert inline code block**:
      Convert function link `|...|` to inline code block `` `...` ``.
      (I've also considered using [`{@linkcode func()}`](https://jsdoc.app/tags-inline-link.html). But it doesn't look good in vim-lsp hover.😭)
- [ ] **Convert emphasis**:
      Convert args `{arg}` to emphasis `__{arg}__`.
      Convert note `{note something ...}` to emphasis `_note something ..._`.

## Others

- [ ] **Detection parse errors**:
      Sometimes it fails to parse silently and is not output.

### PR DESCRIPTION
**Pull request by Milly**
Closes #178

### CODE REVIEW

File: scripts/utils_test.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    () => Promise.resolve(new Response(body, init)),
+  );
+  try {
+    await fn();
+  } finally {
+    fetchStub.restore();
+  }
+}
+
+Deno.test(downloadString.name, async (t) => {
```
Comment (lambdalisue) [inline scripts/utils_test.ts:35]: I didn't think about it but like the idea (using `XXXXX.name` as a test name) 👍

File: scripts/utils.ts [hunk 2]
```diff
@@ -21,7 +21,11 @@ export async function downloadString(url: string): Promise<string> {
  */
 export function regexIndexOf(s: string, pattern: RegExp, offset = 0): number {
   const index = s.slice(offset).search(pattern);
-  return index < 0 ? index : index + offset;
+  if (index < 0 || offset <= -s.length) {
+    return index;
+  } else {
+    return index + offset + (offset < 0 ? s.length : 0);
+  }
```
Comment (lambdalisue) [inline scripts/utils.ts:28]: It would be nice if the behavior for the negative value is documented as the jsdoc.

**Review by lambdalisue — APPROVED** [review]
Nice work!

File: scripts/utils_test.ts (hunk 1 above)
Comment (Milly) [reply scripts/utils_test.ts]: Replaced to string.

File: scripts/utils.ts (hunk 2 above)
Comment (Milly) [reply scripts/utils.ts:29]: Wrote it.

File: scripts/utils_test.ts (hunk 1 above)
Comment (lambdalisue) [reply scripts/utils_test.ts]: Ah... Sorry for the confusion, but I was commenting that I like the way you write your tests.

Comment (Milly) [reply scripts/utils_test.ts:35]: Got it!
Dropped these change. 😄

**Pull request merged**